packages/menu/src/SubMenu.tsx

```diff
@@ -290,66 +299,72 @@ const SubMenu: ExtendableBox<
+                [activeDescriptionTextStyle]: active,
+                [disabledTextStyle]: disabled,
+                [focusStyles.descriptionStyle]: showFocus,
+                [linkDescriptionTextStyle]: typeof rest.href === 'string',
+              })}
+            >
+              {description}
+            </div>
+          )}
         </div>
       </>
     );
 
-    const renderBox =
-      typeof rest.href === 'string' ? (
-        <Box as="a" {...sharedBoxProps}>
-          {boxContent}
-        </Box>
-      ) : (
-        <Box as="button" {...sharedBoxProps}>
-          {boxContent}
-        </Box>
-      );
-
-    const ChevronIcon = open ? ChevronDownIcon : ChevronUpIcon;
-    const chevronIconStyles = cx({
-      [openIconStyle]: open,
-      [closedIconStyle]: !open,
-      [focusedIconStyle]: showFocus,
-    });
+    const as = isAnchor ? 'a' : 'button';
 
     return (
-      <li role="none" className={liStyle}>
-        {renderBox}
-        <IconButton
-          {...iconButton.prop}
-          darkMode={true}
-          ref={setIconButtonElement}
-          aria-label={open ? 'Close Sub-menu' : 'Open Sub-menu'}
-          className={cx(iconButtonStyle, {
-            [openIconButtonStyle]: open,
-            [iconButtonFocusedStyle]: showFocus,
-          })}
-          onClick={(e: React.MouseEvent) => {
-            // we stop the event from propagating and closing the entire menu
-            e.nativeEvent.stopImmediatePropagation();
-
-            if (setOpen) {
-              setOpen(!open);
-            }
-          }}
+      <li role="none">
+        <Box
+          as={as}
+          {...boxProps}
+          {...anchorProps}
+          {...rest}
+          className={cx(
+            menuItemContainerStyle,
+            menuItemHeight['default'],
+            linkStyle,
+            subMenuStyle,
+            {
+              [activeMenuItemContainerStyle]: active,
+              [disabledMenuItemContainerStyle]: disabled,
+              [focusedMenuItemContainerStyle]: showFocus,
+              [subMenuOpenStyle]: open,
+            },
+            className,
+          )}
         >
-          <ChevronIcon
-            role="presentation"
-            className={chevronIconStyles}
-            size={14}
-          />
-        </IconButton>
+          {content}
+          <IconButton
+            {...iconButton.prop}
+            data-testid="lg-sub-menu-icon-button"
+            darkMode={true}
+            ref={setIconButtonElement}
+            aria-label={open ? 'Close Sub-menu' : 'Open Sub-menu'}
+            className={cx(iconButtonStyle, {
+              [openIconButtonStyle]: open,
+              [iconButtonFocusedStyle]: showFocus,
+            })}
+            onClick={(e: React.MouseEvent) => {
+              // we stop the event from propagating and closing the entire menu
+              e.nativeEvent.stopImmediatePropagation();
+
+              if (setOpen) {
+                setOpen(!open);
```

Comment:
No, `setOpen` has the signature `(value: boolean) => void`

Reply:
That's weird that that's the signature, because `setOpen` on the Menu component has the signature `React.Dispatch<React.SetStateAction<boolean>>`, which is what I expected here. Should the type signature just be changed?

Reply:
You'd still call that type the same way. That type just means it was created by calling `useState`.

e.g.
```
const [open,  setOpen] = useState<boolean>(false)
...
setOpen(true) // type is `React.Dispatch<React.SetStateAction<boolean>>`
```

Reply:
Yeah I understand that. What I'm saying is that the type of `setOpen` on SubMenu should probably match the signature of `setOpen` on Menu.

When it comes to a state dispatch call that depends on the previously stored value, it's recommended that we call it in the form of `setOpen(open => !open`. I've found a quick [thread](https://stackoverflow.com/questions/40359800/how-to-toggle-boolean-state-of-a-react-component#answer-48394541) that also links out to React documentation.

Reply:
I agree that the props for `SubMenu`, `MenuItem` and `Menu` should be more unified. That'll require some refactoring of the types that's a bit out of scope for this work. I have a draft PR that starts this refactor in #1176